### PR TITLE
Data layer: list blog stickers

### PR DESCRIPTION
**Pull request by bluefuton**
Connect up the following endpoint to the data layer:

- /sites/%s/blog-stickers

This allows us to list blog stickers for a blog. This forms the basis of a new UI feature for superadmins, to be completed in a later PR.

### To test

`npm run test-client client/state/data-layer/wpcom/sites/blog-stickers`

`npm run test-client client/state/sites/blog-stickers`

### CODE REVIEW

File: client/state/data-layer/wpcom/sites/blog-stickers/index.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+			apiVersion: '1.1',
+			onSuccess: action,
+			onFailure: action,
+		} ),
+	);
+}
+
+export function receiveBlogStickerList( store, action, next, response ) {
+	// validate that it worked
+	if ( ! response || ! isArray( response ) ) {
```
Comment (bluefuton) [inline client/state/data-layer/wpcom/sites/blog-stickers/index.js]: All valid responses from this endpoint return an array:

screen shot 2017-06-15 at 17 42 21

File: client/state/data-layer/wpcom/sites/blog-stickers/index.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		receiveBlogStickerListError( store, action, next );
+		return;
+	}
```
Comment (samouri) [inline client/state/data-layer/wpcom/sites/blog-stickers/index.js]: I may be missing something, but shouldn't this func be doing something in the happy case? like redispatching ?

File: client/state/sites/blog-stickers/actions.js [hunk 3]
```diff
@@ -22,3 +26,12 @@ export function removeBlogSticker( blogId, stickerName ) {
 		},
 	};
 }
+
+export function listBlogStickers( blogId ) {
+	return {
+		type: SITES_BLOG_STICKER_LIST,
+		payload: {
```
Comment (samouri) [inline client/state/sites/blog-stickers/actions.js]: is this pretty?

**Comment by samouri** [comment]
From experience, its pretty hard to correctly test out the data-layer before its actually in use.
I'm all for not-thoroughly testing this at the current stage since its effectively all dead code until it gets hooked up (doesn't even get shipped/bundled in any way)

File: client/state/sites/blog-stickers/actions.js (hunk 3 above)
Comment (bluefuton) [inline client/state/sites/blog-stickers/actions.js]: Yep! JSPrettier in Sublime is claiming "file already formatted".

File: client/state/data-layer/wpcom/sites/blog-stickers/index.js (hunk 2 above)
Comment (bluefuton) [inline client/state/data-layer/wpcom/sites/blog-stickers/index.js]: I'm pretty new to the data layer, but isn't the call to `next` implicit?

I based it on @blowery's data layer work for post email subscriptions:

https://github.com/Automattic/wp-calypso/blob/master/client/state/data-layer/wpcom/read/site/post-email-subscriptions/new/index.js

Comment (samouri) [inline client/state/data-layer/wpcom/sites/blog-stickers/index.js]: right you are. just need to have the reducer listen to the same action

**Pull request merged**